### ISSUE TITLE
Model not working with generators

### ISSUE THREAD

**Issue by garytryan** [thomseddon/koa-oauth-server#41]
using version: 1.0.1

Following the examples exactly, even resorted to just copy/pasting. Doesn't seem to work at all when using generators for the model. Switched to functions, that take a callback and everything is fine.

**Comment by lan-nguyen91** [comment]
Correct, if you want to use async/promise base manner using yield/await in generator, you have to run with babel. Migration of the code from 2.x to 3.x mentioned about this in the documentation. 
https://github.com/oauthjs/node-oauth2-server/wiki/Migrating-from-2.x-to-3.x

**Comment by garytryan** [comment]
Right, got it. Thanks @lan-nguyen91
